modules/infinite-scroll/infinity.js

```diff
@@ -38,19 +38,24 @@
 		this.currentday = settings.currentday;
 		this.order = settings.order;
 		this.throttle = false;
-		this.handle =
-			'<div id="infinite-handle"><span><button>' +
-			text.replace( '\\', '' ) +
-			'</button></span></div>';
 		this.click_handle = settings.click_handle;
 		this.google_analytics = settings.google_analytics;
 		this.history = settings.history;
 		this.origURL = window.location.href;
-		this.pageCache = {};
```

Comment:
**Blocker:** Removing page caching is a major change, which needs to be further analyzed and discussed.

I've consolidated all changes to `pageCache` to a single commit to make is simple for us to revert them if necessary. Please read the [commit message](https://github.com/Automattic/jetpack/pull/15017/commits/57897fdb6a1af1889309600297002e9a66cae691), it contains more details about why this change is being made.

While I understand the reason for implementing the cache the way it's implemented right now, I wasn't able to verify it's actually efficient. Removed nodes from the invisible "pages" from DOM currently don't _seem_ to be garbage collected. I've profiled the memory using Chrome Dev tools and the number of Nodes in memory stayed constant even as they were removed from DOM. I'm not sure why that is, these are my two theories:

1. My machine has enough memory and doesn't need to GC aggressively.
2. The Nodes remain to be referenced somewhere even after they are removed (likely in jQuery internally).

I'll try to construct a more representative test page with a lot of dynamically loaded content and try to verify whether memory can actually be freed by dynamically removing Nodes like this.